src/components/details/CharacterDetails.js

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { useParams } from 'react-router-dom';
+import { useCharacter } from '../../state/characters';
+
+const CharacterDetails = () => {
+  const { id } = useParams();
+  const character = useCharacter(id);
+  if (!character) return <h1>Loading...</h1>;
```

Comment:
this works. you could also add a loading state in your useCharacters hook and bring it in.